Add Pop CLI to Polkadot Docs – Quickstart Development Section

AlexD10S opened the pull request:
This is a draft PR to explore how [Pop CLI](https://onpop.io/cli/) could be integrated into the Polkadot docs to support quick and easy parachain development.

It introduces a new section under Toolkit → Parachain → Quickstart Parachain Development

The goal is to help developers get started faster by:

- Bootstrapping projects with one command

- Setting up their environment automatically

- Running local networks for testing with minimal setup

I opened an issue explaining the proposal and the reasoning behind these changes: https://github.com/polkadot-developers/polkadot-docs/issues/539

This is just a starting point, happy to collaborate and adjust anything to align with the docs style and standards!

nhussein11 reviewed (commented):
@AlexD10S thank you very much! I left some comments but overall this is great!

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,74 @@
+pop build --release
+```
+
+Pop CLI integrates the [Zombienet-SDK](https://github.com/paritytech/zombienet-sdk) allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run:
````
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+This command provides an interactive experience that checks for all necessary dependencies and installs them for you. It’s the fastest and easiest way to get your development environment ready for building parachains with Pop CLI.
+
+--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/install.html'
+
```
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+Developers can quickly begin coding and testing, significantly reducing setup overhead.
+
+
+### Install Pop CLI
```
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+### Install Pop CLI
+
+To install Pop CLI run the following command
```
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,74 @@
+```bash
+cargo install --force --locked pop-cli
+```
+Confirm that Pop CLI is installed by running `pop --help` in your terminal
````
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+### Set up your development environment
+
+To develop and build Polkadot SDK-based it's essential to prepare your local environment with the necessary tools and dependencies. The [Install Polkadot SDK Dependencies](https://docs.polkadot.com/develop/parachains/install-polkadot-sdk/) guide walks you through this setup step-by-step.
```
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,74 @@
+```bash
+pop install
+```
+This command provides an interactive experience that checks for all necessary dependencies and installs them for you. It’s the fastest and easiest way to get your development environment ready for building parachains with Pop CLI.
````
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+## Introduction
+
+Pop CLI is a powerful command-line tool designed explicitly for rapid parachain development within the Polkadot ecosystem. It addresses essential developer needs by providing streamlined commands to set up development environments, scaffold parachain templates, and manage local blockchain networks.
```
Nice! Included a link

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/new-parachain.html'
+
+Once the project is generated, move into the new directory and build your parachain:
```
Solved in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/770862091934e615ec9bc1b997cc1b4f406f941b

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,74 @@
+pop --help
+```
+
+### Set up your development environment
````
Fixed here in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1 and reviewed the rest of new titles are consistent.

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,74 @@
+
+Start a new project quickly using Pop CLI `pop new parachain` command:
+
+--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/new-parachain.html'
```
Nice suggestion. Included an explanation in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1

AlexD10S commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,74 @@
+Once the project is generated, move into the new directory and build your parachain:
+```
+cd my-parachain
+pop build --release
````
Nice suggestion. Included an explanation in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/9f1f9d36cffe27b4f9fc52184e4d3ea16e76daf1

AlexD10S commented:
> @AlexD10S thank you very much! I left some comments but overall this is great!

Apologies for the delay, I was offline for a few days. I've addressed your feedback, and it's looking better now. Thanks for the suggestions!

nhussein11 commented on develop/toolkit/parachains/quickstart/.pages:
```diff
@@ -0,0 +1,5 @@
+nav:
+  - index.md
+  - 'Pop CLI': pop-cli.md
+  
```
```suggestion
```

AlexD10S replied:
Fixed https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/8202328458758e4e219784f4f6522229dfde53f2

nhussein11 commented on develop/toolkit/parachains/quickstart/index.md:
```diff
@@ -0,0 +1,13 @@
+---
+title: Quickstart Development
+description: Quickly bootstrap your development environment, scaffold template-based projects, deploy local networks, and simplify your workflow with intuitive tools designed specifically for parachain developers.
+
```
```suggestion
```

AlexD10S replied:
Fixed https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/8202328458758e4e219784f4f6522229dfde53f2

nhussein11 commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,84 @@
+```
+
+!!! note
+    Under the hood, `pop build --release` runs `cargo build --release`, but `pop build` adds functionality specific to Polkadot SDK projects, such as deterministic runtime builds and automatic management of feature flags like `benchmark` or `try-runtime`.
````
We do have a page discussing deterministic runtimes - https://docs.polkadot.com/develop/parachains/deployment/build-deterministic-runtime/. @AlexD10S, if you think adding this link to the paragraph would be helpful, feel free to do so. If not, you can just ignore it

```suggestion
    Under the hood, `pop build --release` runs `cargo build --release`, but `pop build` adds functionality specific to Polkadot SDK projects, such as [deterministic runtime builds](/develop/parachains/deployment/build-deterministic-runtime/){target=\_blank} and automatic management of feature flags like `benchmark` or `try-runtime`.
```

nhussein11 reviewed (approved):
LGTM, just left comments about some minor details! From a content perspective, I think this PR is good to go. @eshaben @dawnkelly09 @0xlukem — would you mind giving it a quick review?

If you notice quite a few formatting tweaks are needed, feel free to tag me — I can merge this into a separate branch, work on the adjustments, and open a follow-up PR. Otherwise, let’s go ahead and have Alex wrap up his contribution.

eshaben commented on develop/toolkit/parachains/quickstart/index.md:
```diff
@@ -0,0 +1,11 @@
+---
+
+# Quickstart Parachain Development
+
```
We should consider some content that might be helpful for this section. I don't know what else the plan is for this section, like what other pages are going to be added here?

eshaben replied:
Ok, no worries here. Thanks!

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,84 @@
+
+### Set Up Your Development Environment
+
+To develop and build Polkadot SDK-based it's essential to prepare your local environment with the necessary tools and dependencies. The [Install Polkadot SDK Dependencies](/develop/parachains/install-polkadot-sdk/){target=\_blank} guide walks you through this setup step-by-step.
```
```suggestion
To develop and build Polkadot SDK-based chains, preparing your local environment with the necessary tools and dependencies is essential. The [Install Polkadot SDK Dependencies](/develop/parachains/install-polkadot-sdk/){target=\_blank} guide walks you through this setup step-by-step.
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
````diff
@@ -0,0 +1,84 @@
+pop install
+```
+
+This command provides an interactive experience that checks for all necessary dependencies and installs them for you. It’s the fastest and easiest way to get your development environment ready for building parachains with Pop CLI.
````
```suggestion
This command provides an interactive experience that checks and installs all necessary dependencies for you. It’s the fastest and easiest way to prepare your development environment for building parachains with Pop CLI.
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,84 @@
+
+### Initialize a Project
+
+Start a new project quickly using Pop CLI `pop new parachain` command:
```
```suggestion
Start a new project quickly using Pop CLI's `pop new parachain` command:
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,84 @@
+
+--8<-- 'code/develop/toolkit/parachains/quickstart/pop-cli/new-parachain.html'
+
+The command above scaffolds a new parachain project using the default template included with Pop CLI. For more specialized implementations, additional templates are available, you can explore them by running `pop new parachain --help`.
```
```suggestion
The command above scaffolds a new parachain project using the default template included with Pop CLI. For more specialized implementations, additional templates are available; you can explore them by running `pop new parachain --help`.
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,84 @@
+!!! note
+    Under the hood, `pop build --release` runs `cargo build --release`, but `pop build` adds functionality specific to Polkadot SDK projects, such as [deterministic runtime builds](/develop/parachains/deployment/build-deterministic-runtime/){target=\_blank} and automatic management of feature flags like `benchmark` or `try-runtime`.
+
+Pop CLI integrates the [Zombienet-SDK](https://github.com/paritytech/zombienet-sdk){target=\_blank} allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run the following:
```
```suggestion
Pop CLI integrates the [Zombienet SDK](https://github.com/paritytech/zombienet-sdk){target=\_blank}, allowing you to easily launch ephemeral local networks for development and testing. To start a network, simply run the following:
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben commented on develop/toolkit/parachains/quickstart/pop-cli.md:
```diff
@@ -0,0 +1,84 @@
+
+This command will automatically fetch the necessary binaries and spin up a Polkadot network with your configured parachains.
+
+You can also interact with your local network using Pop CLI `pop call chain` command:
```
```suggestion
You can also interact with your local network using Pop CLI's `pop call chain` command:
```

AlexD10S replied:
Fixed in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben reviewed (commented):
Nice! Just some grammar and SEO-related suggestions 🙂

AlexD10S commented on develop/toolkit/parachains/quickstart/index.md:
```diff
@@ -0,0 +1,11 @@
+---
+title: Quickstart Development
```
Nice feedback, updated title in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

AlexD10S commented on develop/toolkit/parachains/quickstart/index.md:
```diff
@@ -0,0 +1,11 @@
+---
+title: Quickstart Development
+description: Quickly bootstrap your development environment, scaffold template-based projects, deploy local networks, and simplify your workflow with intuitive tools designed specifically for parachain developers.
```
Updated the description in https://github.com/polkadot-developers/polkadot-docs/pull/538/commits/cf6091a8a444f08447db2bdaa88438254130b286

eshaben reviewed (approved):
LGTM! Thanks!

eshaben commented:
Merging this into a local branch to resolve the failing build checks
